dev/com.ibm.ws.concurrent.persistent_fat_multiple/fat/src/com/ibm/ws/concurrent/persistent/fat/multiple/DatabaseContainerFactory.java

```diff
@@ -0,0 +1,144 @@
+	private static String database = System.getProperty("fat.bucket.db.type", "Derby");
+	private static GenericContainer<?> container = initContainer();
+	
+	@SuppressWarnings("rawtypes")
+	private static Class[] classes;
+	
+	/**
+	 * Reads the 'fat.bucket.db.type' system property and returns a
+	 * container based on that property.
+	 * 
+	 * If we are using a network database, a test container running
+	 * that database will be returned.  
+	 * [Postgre, DB2, Oracle, SQLServer, MySQL]
+	 * 
+	 * If we are using derby embedded or unsupported database
+	 * a no-op container will be returned.
+	 * [Derby, Informix, Sybase]
+	 * 
+	 * @param testClasses - Provide the name(s) of your test class(es). These will be passed to the test container which will then provide log output to the class(es) log.  
+	 * @return GenericContainer - Test test container.
+	 */
+	@SuppressWarnings("rawtypes")
+	public static GenericContainer<?> get(Class... testClasses) {
+		classes = testClasses;
+		return container;
+	}
+	
+	/**
+	 * Returns true if the test container is configured
+	 * for a network database.
+	 * This will need to be checked prior to setting
+	 * server environment variables for database connectivity
+	 * @return
+	 */
+	public static boolean isNetworkDatabase() {
+		return container instanceof  JdbcDatabaseContainer;
+	}
+	
+	/**
+	 * Returns the name of the database that will be 
+	 * used to generate a test container. 
+	 * @return String - name of database
+	 */
+	public static String usingDB() {
+		return database;
+	}
+	
+	/**
+	 * Private Method: used to initialize test container.
+	 */
+	private static GenericContainer<?> initContainer() {		
+    	switch(database) {
+    	case "Postgre":
+    		return new PostgreSQLContainer<>()
+    				.withLogConsumer(DatabaseContainerFactory::log);
+		case "DB2":
+			return new Db2Container()
+                    .acceptLicense()
+                    // Use 5m timeout for local runs, 15m timeout for remote runs
+                    .withStartupTimeout(Duration.ofMinutes(FATRunner.FAT_TEST_LOCALRUN ? 5 : 15))
+                    .withLogConsumer(DatabaseContainerFactory::log);
+		case "Oracle":
+			//TODO does IBM/Websphere have an oracle docker container that could be used here instead?
+			return new OracleContainer("oracleinanutshell/oracle-xe-11g")
+					.withLogConsumer(DatabaseContainerFactory::log);
+		case "SQLServer":
+			return new MSSQLServerContainer<>()
+					.withLogConsumer(DatabaseContainerFactory::log);
+		case "Derby":
+		default:
+			return new MockServerContainer();
+    	}
+    }
+	
+	/**
+	 * Returns port to be used to access test container.
+	 * Each database uses a different internal port to accept incoming 
+	 * connection requests.  The test container automatically maps a 
+	 * random external port to the default internal port. 
+	 * 
+	 * @return int - port of test container to be used for database access.
+	 */
+	public static int getMappedPort() {
```

Comment:
I don't this this method is necessary, instead the test can just call `container.getFirstMappedPort()`